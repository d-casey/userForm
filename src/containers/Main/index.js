import React from 'react'
import Sidebar from '../../components/Sidebar'
import Router from '../../components/Router'
import ProgressBar from '../../components/ProgressBar'
import { useSelector } from "react-redux"
import mainContainer from '../../css/mainContainer.css'

function App() {
  const appStage = useSelector(state => state.app.stage)

  return (
    <div className="main-container">
      <ProgressBar appStage={appStage} />
      <div className="wrapper">
        <Sidebar appStage={appStage} />
        <Router />
      </div>
    </div>
  )
}

export default App
