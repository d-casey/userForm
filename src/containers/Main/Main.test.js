import React from 'react'
import Main from '.'
import { Provider } from 'react-redux'

import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
const mockStore = configureStore([])

let store
let component
let initialState = {
  app: {
    stage: 0,
    formCompleted: true
  },
  newUserForm: {
    name: "",
    role: "",
    email: "",
    password: "",
    isSubmitted: false
  },
  privacyForm : {
    isSubmitted: false,
    receive_updates: true,
    receive_communication: false,
  }
}

const getWrapper = (state) => {
  store = mockStore(state)
  component = renderer.create(
    <Provider store={store}>
      <Main />
    </Provider>
  )
}

describe('<Main /> container', () => {
  describe('When the app stage is 0', () => {
    beforeEach(() => {
      getWrapper(initialState)
    })

    it('passes the state into the sidebar and progress bar', () => {
      expect(component.toJSON()).toMatchSnapshot()
    })
  })

  describe('When the app stage is 1', () => {
    beforeEach(() => {
      let updatedState = initialState
      updatedState.app.stage = 1
      getWrapper(updatedState)
    })

    it('should render the correct sidebar and progress bar content', () => {
      expect(component.toJSON()).toMatchSnapshot()
    })
  })

  describe('When the app stage is 2', () => {
    beforeEach(() => {
      let updatedState = initialState
      updatedState.app.stage = 2
      getWrapper(updatedState)
    })

    it('should render the correct sidebar and progress bar content', () => {
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
