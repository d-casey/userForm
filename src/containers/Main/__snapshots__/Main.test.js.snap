// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> container When the app stage is 0 passes the state into the sidebar and progress bar 1`] = `
<div
  className="main-container"
>
  <div
    className="progress-bar"
  >
    <ul
      className="progress-bar-flex-container"
    >
      <li
        className="progress-bar-flex-item active-item"
      >
        User
      </li>
      <li
        className="progress-bar-flex-item"
      >
        Privacy
      </li>
      <li
        className="progress-bar-flex-item"
      >
        Done
      </li>
    </ul>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="sidebar"
    >
      <h3>
        Sign Up Form
      </h3>
      <p>
        Required fiels are marked with an *
      </p>
      <p>
        Passwords must be:
      </p>
      <ul>
        <li>
          At least 10 characters long
        </li>
        <li>
          Contain at least one number
        </li>
        <li>
          Contain at least one upper case letter
        </li>
        <li>
          Contain at least one lower case letter
        </li>
      </ul>
    </div>
    <div
      className="form"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Name
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="name"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Role
            : 
            
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="role"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Email
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="email"
            type="email"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Password
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            name="password"
            type="password"
          />
        </fieldset>
        <button
          className="right"
          type="submit"
        >
          Next
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<Main /> container When the app stage is 1 should render the correct sidebar and progress bar content 1`] = `
<div
  className="main-container"
>
  <div
    className="progress-bar"
  >
    <ul
      className="progress-bar-flex-container"
    >
      <li
        className="progress-bar-flex-item"
      >
        User
      </li>
      <li
        className="progress-bar-flex-item active-item"
      >
        Privacy
      </li>
      <li
        className="progress-bar-flex-item"
      >
        Done
      </li>
    </ul>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="sidebar"
    >
      <h3>
        Privacy
      </h3>
      <p>
        Please let us know how you would like us to contact you.
      </p>
    </div>
    <div
      className="form"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Name
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="name"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Role
            : 
            
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="role"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Email
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="email"
            type="email"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Password
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            name="password"
            type="password"
          />
        </fieldset>
        <button
          className="right"
          type="submit"
        >
          Next
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`<Main /> container When the app stage is 2 should render the correct sidebar and progress bar content 1`] = `
<div
  className="main-container"
>
  <div
    className="progress-bar"
  >
    <ul
      className="progress-bar-flex-container"
    >
      <li
        className="progress-bar-flex-item"
      >
        User
      </li>
      <li
        className="progress-bar-flex-item"
      >
        Privacy
      </li>
      <li
        className="progress-bar-flex-item active-item"
      >
        Done
      </li>
    </ul>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="sidebar"
    >
      <h3>
        Done!
      </h3>
    </div>
    <div
      className="form"
    >
      <form
        onSubmit={[Function]}
      >
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Name
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="name"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Role
            : 
            
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="role"
            type="text"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Email
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            defaultValue=""
            name="email"
            type="email"
          />
        </fieldset>
        <fieldset
          className="full-width-imput"
        >
          <legend>
            Password
            : 
            *
             
            <span
              className="error"
            />
          </legend>
          <input
            name="password"
            type="password"
          />
        </fieldset>
        <button
          className="right"
          type="submit"
        >
          Next
        </button>
      </form>
    </div>
  </div>
</div>
`;
